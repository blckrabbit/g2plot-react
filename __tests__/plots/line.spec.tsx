import React from 'react'
import ReactDOM from 'react-dom'
import { create, act, ReactTestRenderer } from 'react-test-renderer'
import LineChart from '../../src/plots/line'

describe('LineChart', () => {
  test('render without crashed', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LineChart data={[]} />, div)
  })

  test('onMount should be called', () => {
    const onMount = jest.fn()
    const div = document.createElement('div')
    ReactDOM.render(<LineChart data={[]} onMount={onMount} />, div)

    expect(onMount).toBeCalled()
  })

  test('test update config and data', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LineChart data={[]} />, div)

    ReactDOM.render(<LineChart data={[]} forceFit />, div)

    ReactDOM.render(<LineChart data={[{ x: 1 }]} forceFit />, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  test('test lifecycle', () => {
    let renderer: ReactTestRenderer

    act(() => {
      renderer = create(<LineChart data={[]} />)
    })

    const instance = renderer.root

    expect(instance.props.data).toEqual([])

    expect(renderer.toJSON()).toMatchSnapshot()

    act(() => {
      renderer.update(<LineChart data={[{ x: 1 }]} forceFit />)
    })

    expect(renderer.toJSON()).toMatchSnapshot()

    expect(instance.props.data).toEqual([{ x: 1 }])

    act(() => {
      renderer.unmount()
    })
  })
})
