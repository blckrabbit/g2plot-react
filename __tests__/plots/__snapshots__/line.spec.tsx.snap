// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart lifecycle 1`] = `<div />`;
